examples/asynchronous_consumer_example.py

```diff
@@ -48,9 +49,11 @@ def connect(self):
 
         """
         LOGGER.info('Connecting to %s', self._url)
-        return pika.SelectConnection(pika.URLParameters(self._url),
-                                        self.on_connection_open,
-                                        stop_ioloop_on_close=False)
+        return pika.SelectConnection(parameters=pika.URLParameters(self._url),
+                                     on_open_callback=self.on_connection_open,
+                                     on_open_error_callback=self.on_connection_open_error,
+                                     on_close_callback=self.on_connection_closed,
+                                     stop_ioloop_on_close=False)
 
```

Comment:
The new call signature for `pika.connection.channel` is `def channel(self, channel_number=None, callback=None)`. However, the `open_channel` function in the asynchronous consumer example calls it using an incorrect signature: `self._connection.channel(on_open_callback=self.on_channel_open)`

Reply:
There are a number of others like it:
```
$ git grep 'on_open_callback='
docs/examples/asynchronous_consumer_example.rst:            self._connection.channel(on_open_callback=self.on_channel_open)
docs/examples/asynchronous_publisher_example.rst:            self._connection.channel(on_open_callback=self.on_channel_open)
docs/examples/asyncio_consumer.rst:            self._connection.channel(on_open_callback=self.on_channel_open)
docs/examples/tornado_consumer.rst:            self._connection.channel(on_open_callback=self.on_channel_open)
examples/asynchronous_consumer_example.py:        self._connection.channel(on_open_callback=self.on_channel_open)
examples/asynchronous_publisher_example.py:        self._connection.channel(on_open_callback=self.on_channel_open)
examples/confirmation.py:                                   on_open_callback=on_open)
```

Reply:
in docs/examples/comparing_publishing_sync_async.rst, asynchronous `connection.channel` call doesn't match the new `Connection.channel` signature. it's called simply as `connection.channel(on_channel_open)`, which ends up passing the callback as first arg, which is now `channel_number`.

Reply:
I left the callback named as `on_open_callback` so its unambiguous what it does. I'll make sure the docs / examples are all in sync.

Reply:
This should be all set now. I kept the name `on_open_callback` for both connections and channels to be consistent.